### ISSUE TITLE
z80asm: Remove the z80asm INVOKE directive?

### ISSUE THREAD

**Issue by pauloscustodio** [z88dk/z88dk#18]
from:	pscust (pauloscustodio@gmail.com) <lists@suborbital.org.uk> via lists.sourceforge.net 
to:	z88dk-developers@lists.sourceforge.net
date:	Fri, Jul 29, 2016 at 2:05 PM

z80asm implements the directive "INVOKE xx" as "call xx", or as "rst 28h; defw xx" with the --ti83plus option.
This directive is not used in the library, neither is the option --ti83plus.

I propose to remove the implementation of INVOKE from z80asm, as well as the --ti83plus option. I plan to have a macro facility in z80asm, which will allow this kind of platform-specific directives to be implemented in a platform-specific include file.

Any comments?

**Comment by pauloscustodio** [comment]
from:	alvin (alvin_albrecht@hotmail.com) <lists@suborbital.org.uk> via lists.sourceforge.net 
to:	z88dk-developers@lists.sourceforge.net
date:	Thu, Aug 4, 2016 at 7:47 AM

>z80asm implements the directive "INVOKE xx" as "call xx", or as "rst 28h; defw xx" with the --ti83plus option.
>This directive is not used in the library, neither is the option --ti83plus.
>
>I propose to remove the implementation of INVOKE from z80asm, as well as the --ti83plus option. I plan to have a macro facility in z80asm, which will allow this kind of platform-specific directives to be implemented in a platform-specific include file.
I'd say remove it and later replace with a macro.  I think it is a ti-calc specific asm macro convention.  I don't think we have many ti-calc users so a brief-ish disappearance won't be missed I think.

**Comment by pauloscustodio** [comment]
from:	Stefano Bodrato (stefano_bodrato@hotmail.com) <lists@suborbital.org.uk> via lists.sourceforge.net 
to:	z88dk-developers@lists.sourceforge.net
date:	Thu, Aug 4, 2016 at 9:35 AM

I had totally forgotten that courtesy we did to the TI world !
Yes, macros seem a much better way to deal with similar situations.   We have some kind of OZCALL for the z88 too, right ?

**Comment by pauloscustodio** [comment]
from:	Paulo Custodio <pauloscustodio@gmail.com>
to:	z88dk-developers@lists.sourceforge.net
cc:	"Stefano Bodrato (stefano_bodrato@hotmail.com)" <lists@suborbital.org.uk>
date:	Thu, Aug 4, 2016 at 2:32 PM

> I had totally forgotten that courtesy we did to the TI world !
> Yes, macros seem a much better way to deal with similar situations.   We
> have some kind of OZCALL for the z88 too, right ?

Yes, we still have CALL_OZ, CALL_PKG and FPP, but these cannot be
removed until we have macros, as they are used in the library.

We also had OZ as an alias to CALL_OZ that was unused and I did remove.

I have started z80asm2 from scratch in C++ with the goal to replace
z80asm and add new features. Macros and link memory map control are in
my first priorities. It is rather far from finished though. In the
mean time I'll maintain and simplify whenever possible z80asm.

I also need to have a look at z80asm.txt - it is completely out-dated.
